src/potions.ts

```diff
@@ -0,0 +1,211 @@
+    if (keepGoing && this.net(embezzlersRemaining, doubleDuration) > 0 && embezzlersRemaining > 0) {
+      acquire(1, this.potion, this.gross(embezzlersRemaining, doubleDuration), false);
+      const additional = Math.max(1, itemAmount(this.potion) - quantityToUse);
+      print(
+        `Determined that ${additional} ${this.potion.plural} are profitable on partial embezzlers.`,
+        "blue"
+      );
+      quantityToUse += additional;
+      embezzlersRemaining = Math.max(embezzlersRemaining - additional * duration, 0);
+    }
+    if (embezzlersRemaining > 0 || (doubleDuration && quantityToUse > 0)) keepGoing = false;
+
+    // How many should we use with non-embezzlers?
+    if (keepGoing && this.net(0, doubleDuration) > 0) {
+      const adventureCap = myAdventures() * 1.04 + 50;
+      const tourists = adventureCap - haveEffect(this.effect()) - quantityToUse * duration;
+      if (tourists > 0) {
+        const touristQuantity = Math.ceil(tourists / duration);
+        acquire(touristQuantity, this.potion, this.gross(0, doubleDuration), false);
+        const additional = Math.max(touristQuantity, itemAmount(this.potion) - quantityToUse);
+        print(
+          `Determined that ${additional} ${this.potion.plural} are profitable on tourists.`,
+          "blue"
+        );
+        quantityToUse += additional;
+      }
+    }
+
+    if (quantityToUse > 0) {
+      if (doubleDuration) quantityToUse = 1;
+      use(quantityToUse, this.potion);
+    }
+  }
+}
+
+export function potionSetup(): void {
+  // TODO: Count PYEC.
+  // TODO: Count free fights (25 meat each for most).
+  const embezzlers = embezzlerCount();
+  const potions = Item.all().filter(
+    (item) => item.tradeable && !banned.includes(item) && itemType(item) === "potion"
+  );
+  const meatPotions = potions
+    .map((item) => new Potion(item))
+    .filter((potion) => potion.bonusMeat() > 0);
+
+  if (have($item`Eight Days a Week Pill Keeper`) && !get("_freePillKeeperUsed")) {
+    const testPotionsDoubled = meatPotions.filter(
+      (potion) => potion.gross(embezzlers, true) / potion.price(true) > 0.5
+    );
+    testPotionsDoubled.sort((x, y) => -(x.doublingValue(embezzlers) - y.doublingValue(embezzlers)));
+    for (const potion of testPotionsDoubled) {
+      print(`DOUBLE ${potion.potion.name}: ${potion.doublingValue(embezzlers).toFixed(0)}`);
+    }
+    if (testPotionsDoubled.length > 0) {
+      const potion = testPotionsDoubled[0];
+      // Estimate that the opportunity cost of free PK useage is 10k meat - approximately +1 embezzler.
+      if (potion.doublingValue(embezzlers) > 10000) {
+        cliExecute("pillkeeper extend");
+        potion.useAsValuable(embezzlers, true);
+      }
+    }
+  }
+
+  // Only test potions which are reasonably close to being profitable using historical price.
+  const testPotions = meatPotions.filter(
+    (potion) => potion.gross(embezzlers) / potion.price(true) > 0.5
```

Comment:
Should this be > 1? Should we use subtraction comparing to 0 instead of division?

Reply:
This is trying to do a rough filtering using historical price, and then do a precise check later using current price. So it checks to see if it's within a factor of 2 of being profitable.